plenum/server/replica.py

```diff
@@ -1635,28 +1771,58 @@ def send(self, msg, stat=None) -> None:
         self.outBox.append(msg)
 
     def revert_unordered_batches(self, ledger_id):
+        i = 0
         for key in sorted(self.batches.keys(), reverse=True):
-            if key > self.lastOrderedPPSeqNo:
+            if compare_3PC_keys(self.last_ordered_3pc, key) > 0:
                 count, _, prevStateRoot = self.batches.pop(key)
                 self.revert(ledger_id, prevStateRoot, count)
+                i += 1
             else:
                 break
+        return i
 
-    def caught_up_till_pp_seq_no(self, last_caught_up_pp_seq_no):
-        self.addToOrdered(self.viewNo, last_caught_up_pp_seq_no)
```

Comment:
`last_ordered_3pc` is set since `self.ordered` is not accurate anyway since if you are getting 1000 txns from catchup, you don't know how many batches got ordered but only the last batch, its not safe to calculate number of batches in between if there were view changes involved